web/src/copy_and_paste.ts

```diff
@@ -472,6 +489,122 @@ export function paste_handler_converter(paste_html: string): string {
             return prefix + content + (node.nextSibling && !content.endsWith("\n") ? "\n" : "");
         },
     });
+    const processed_math_block_parents = new Set();
+    const processed_inline_math_block_parents = new Set();
+    turndownService.addRule("katex-math-block", {
+        filter(node: Node) {
+            if (!(node instanceof HTMLElement)) {
+                return false;
+            }
+            // We only need nodes whose parent contains the katex-display spans as immediate children or katex spans as immediate children
+            if (
+                node.classList.contains("katex-display") &&
+                !node.closest(".katex-display > .katex-display")
+            ) {
+                return true;
+            }
+
+            return false;
+        },
+        replacement(content: string, node: Node) {
+            assert(node instanceof HTMLElement);
+            let math_block_markdown = "```math\n";
+            if (node !== null) {
+                const parent = node.parentElement!;
+                if (processed_math_block_parents.has(parent)) {
+                    return "";
+                }
+                processed_math_block_parents.add(parent);
```

Comment:
Some comments should explain what you're doing here with the `processed_math_block_parents` set -- I get that you're trying to process each something once, but it's kinda unclear why you're just not relying on the `filter` bit above to pass you each math block once...